scraper.go

```diff
@@ -132,38 +133,102 @@ func done(ctx context.Context) bool {
 // worker is a function to process roms from a channel.
 func worker(ctx context.Context, sources []ds.DS, xmlOpts *rom.XMLOpts, gameOpts *rom.GameOpts, results chan result, roms chan *rom.ROM, wg *sync.WaitGroup) {
 	defer wg.Done()
-	for r := range roms {
+
+	// TODO(jpr): I think theres probably a way to dedicate a step in the pipeline
+	// to batching, but I'm not sure how to handle retries safely in the batcher
```

Comment:
maybe something like creating a "job" struct. It contains the rom, and retries attempted, etc. Then the worker or batcher can discard(or send it to the next data source's batcher) the job after it exceeds the retries.